sled-agent/src/services.rs

```diff
@@ -773,6 +773,100 @@ pub struct ServiceManager {
     inner: Arc<ServiceManagerInner>,
 }
 
+/// Ensure that a NAT entry exists, overwriting a previous conflicting entry if
+/// applicable.
+///
+/// nat_ipv\[46\]_create are not idempotent (see oxidecomputer/dendrite#343),
+/// but this wrapper function is. Call this from sagas instead.
+#[allow(clippy::too_many_arguments)]
```

Comment:
Worth creating a type to avoid the clippy?

Reply:
I generally don't bother unless a function is called from more than one place or from another file/module, but I can if you'd like.

Reply:
Nah, I'm not a stickler.